cmd/hil-vpnd/vpnstates.go

```diff
@@ -34,12 +34,33 @@ type VpnStates struct {
 // methods are not (but may be called when the VpnStates is already
 // locked).
 
-// Allocate an empty VpnStates.
-func newStates() *VpnStates {
-	return &VpnStates{
+// Allocate a fresh VpnStates, with `FreePorts` generated based on
+// the config. The `vpnNames` argument should the output of
+// PrivOps.ListVPNs.
+func newStates(cfg config, vpnNames []string) *VpnStates {
+	ret := &VpnStates{
 		UsedPorts: map[UniqueId]uint16{},
 		FreePorts: []uint16{},
 	}
+	usedPorts := make(map[uint16]struct{})
+	for _, name := range vpnNames {
+		id, port, err := parseVpnName(name)
+		if err != nil {
+			// skip it; perhaps the local sysadmin created an
```

Comment:
Is it assumed that any malformed hil-vpn names wouldn't have made it in in the first place, so any errors here are something else?